packages/hooks-react/src/useCheckAccess.ts

```diff
@@ -23,20 +23,28 @@ const useCheckAccess = () => {
 
   const intervalCheckAccess = useCallback(
     ({ interval = 3000, iterations = 5, offerId, callback }: IntervalCheckAccessPayload) => {
-      if (!offerId && offers?.[0]) {
-        offerId = offers[0];
+      if (!offerId) {
+        offerId = offers?.[0] || '';
+      }
+
+      if (!offerId) {
+        callback?.({ hasAccess: false, offerId: '' });
+        return;
```

Comment:
Could we default the arg? 
```javascript
({ interval = 3000, iterations = 5, offerId = offers?.[0], callback }: IntervalCheckAccessPayload) => {
```

Or otherwise perhaps:
```suggestion
      if (!offerId) {
        if (!offers?.[0]) {
          callback?.({ hasAccess: false, offerId: '' });
          
          return;
        }
        offerId = offers[0];
      }
```

Reply:
Yes, good suggestion!